### ISSUE TITLE
The count of <MASK> token generated is less than 16 or even equals to zero.

### ISSUE THREAD

**Issue by xiangwentao666** [nnnth/UFO#29]
1. Could you please tell me what to do if this problem occurs?
2. For the code below, if the ```original_shape``` is ```(448, 448)```, is it valid and correct to change ```8``` to ```448``` through ```interpolate``` operation?

The shape of tensor ```img_feat``` is ```[bs, 256, 4096]``` and ```img_feat``` is extracted from the final layer output sequence of LLM. 

Meanwhile, assuming the shape of mask_feat to be ```[bs x 16, 4096]``` here, the shape of ```pred_mask``` should be [bsx16, bsx256], and the ```height``` and ```width``` variables would be ```bs``` and ```256```. 

So, when interpolating the pred_mask to original shape of the input images, the shape of ```pred_mask``` would change from ```[2, 8, 1024]``` to ```[2, *original_shape]```. 

```python
        if len(mask_feat) > 0:
            mask_feat = torch.stack(mask_feat,dim=0).flatten(0,1)
            assert mask_feat.shape[0] % 16 == 0

            pred_mask = mask_feat @ img_feat.flatten(0,1).permute(1,0) / math.sqrt(mask_feat.shape[-1])
            height,width = img_feat.shape[:2]
            pred_mask = pred_mask.view(mask_feat.shape[0], height, width)
            pred_mask = pred_mask.view(mask_feat.shape[0]//16, 4, 4, height, width).permute(0, 3, 1, 4, 2).flatten(1,2).flatten(2,3)

            pred_mask = F.interpolate(
                pred_mask.unsqueeze(1),
                original_shape,
                mode='bilinear',
                align_corners=False,
            ).squeeze(1)
            pred_mask = pred_mask.sigmoid()
```

**Comment by xiangwentao666** [comment]
> 1. If the output `<MASK>` tokens during testing do not satisfy 16, I will consider the mask prediction to have failed and output an empty mask. However, this situation is rare and mostly occurs in the early stages of training when the model has not yet converged.
> 2. In the code shown above, the `img_feat` corresponds to the features of a single image with the shape (256, 4096). The `mask_feat` has features in the shape (Nx16, 4096), where N is the number of masks for the image. For example, in instance segmentation, an image may have multiple masks, while in refer segmentation, there is only one mask per image. Therefore, the shape of `pred_mask` is (Nx16, 256), which is then reshaped to (Nx16, 256) -> (N, 4, 4, 16, 16) -> (N, 4, 16, 4, 16) -> (N, 64, 64). Each of the 16 tokens corresponds to a fixed position in the image. If the number of predicted `<MASK>` tokens does not match the number set in training, the position correspondences are also invalid. If you must output a mask anyway, there are two optional methods:
> 
> * Use only the first `<MASK>` token, making the shape of the `pred_mask` (N, 16, 16), which may provide a rough output.
> * Assume the K predicted `<MASK>` tokens are the first K of the intended 16 tokens, and you can pad the predicted mask from (NxK, 256) to (Nx16, 256), potentially yielding valid output.
> 
> In summary, this situation should not occur; it might be due to the model not converging during training, or perhaps the prompt does not explicitly include hints such as "Please output segmentation mask."


My prompt text is ```'system\n你是由上海人工智能实验室联合商汤科技开发的书生多模态大模型，英文名叫InternVL, 是一个有用无害的人工智能助手。user\nWhat is "a man on the train watching the rain outside" in this image? Please output segmentation mask.\nassistant\n'```，and I am using your publicly released pre-trained weights and trying to decode more than one sample at a time during inference (i.e., batch_size > 1). Could you please:
(1) help me check if there is any issue with the prompt?
(2) tell me whether only one sample is allowed or not when I am decoding masks?
